### DISCUSSION TITLE
gunicorn randomly getting term when running fastapi

### DISCUSSION THREAD

**Discussion by jmugan** [fastapi/fastapi#10665] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
[Unit]
Description=Gunicorn FastAPI APP

[Service]
User=ubuntu
Group=ubuntu
Restart=always
WorkingDirectory=/home/ubuntu/billenote
ExecStop=/home/ubuntu/scripts/kill_gunicorn.sh
TimeoutStopSec=5
Environment="PYTHONPATH=/home/ubuntu/billenote/src"
ExecStart=/home/ubuntu/anaconda3/bin/gunicorn src.app.main:app \
  -w 2 -k uvicorn.workers.UvicornWorker --bind 127.0.0.1:8000 --timeout 90

[Install]
WantedBy=multi-user.target
```

### Description

I'm running gunicorn in production with systemd. I periodically get `[INFO] Handling signal: term` and then `Error while closing socket [Errno 9] Bad file descriptor` and then it won't start up again saying `Connection in use: ('127.0.0.1', 8000)`. This has been driving me nuts because prod goes down. Any help or pointers are appreciated.

I running with with FastAPI and I'm on gunicorn 21.2.0.

And I checked memory, that's not it.

### Operating System

Linux

### Operating System Details

Ubuntu

### FastAPI Version

0.72.0

### Pydantic Version

1.10.12

### Python Version

Python 3.8.8

### Additional Context

This is more about running FastAPI in production than FastAPI itself. Are there any alternatives I should be using to gunicorn?

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
